File: src/main/java/me/muksc/tacztweaks/client/input/UnloadKey.java
package me.muksc.tacztweaks.client.input;

import com.mojang.blaze3d.platform.InputConstants;
import me.muksc.tacztweaks.TaCZTweaks;
import me.muksc.tacztweaks.TaCZTweaksClient;
import me.muksc.tacztweaks.config.Config;
import me.muksc.tacztweaks.network.NetworkHandler;
import net.minecraft.client.KeyMapping;
import org.lwjgl.glfw.GLFW;

public class UnloadKey {
    public static final KeyMapping KEY = new KeyMapping(
        TaCZTweaks.translatable("key.unload").getString(),
        InputConstants.Type.KEYSYM,
        GLFW.GLFW_KEY_U,
        TaCZTweaksClient.CATEGORY
    );

    public static void onClientTick() {
        if (!Config.Gun.INSTANCE.allowUnload()) return;
        while (KEY.consumeClick()) {
            NetworkHandler.INSTANCE.sendPlayerUnload();
        }
    }
}

File: src/main/kotlin/me/muksc/tacztweaks/client/input/UnloadKey.kt
package me.muksc.tacztweaks.client.input

import com.mojang.blaze3d.platform.InputConstants
import me.muksc.tacztweaks.TaCZTweaks
import me.muksc.tacztweaks.config.Config
import me.muksc.tacztweaks.network.NetworkManager
import me.muksc.tacztweaks.network.message.ClientMessagePlayerUnload
import net.minecraft.client.KeyMapping

object UnloadKey {
    @JvmField
    val KEY = KeyMapping(
        TaCZTweaks.key("unload"),
        InputConstants.UNKNOWN.value,
        ModKeyCategory.VALUE
    )

    fun onEndClientTick() {
        if (!Config.KeyActions.Unload.enabled()) return
        while (KEY.consumeClick()) {
            NetworkManager.sendC2S(ClientMessagePlayerUnload)
        }
    }
}